Mathlib/Analysis/Normed/Group/SemiNormedGroup.lean

```diff
@@ -0,0 +1,307 @@
+#align SemiNormedGroup SemiNormedGroup
+
+namespace SemiNormedGroup
+
+instance bundledHom : BundledHom @NormedAddGroupHom where
+  toFun := @NormedAddGroupHom.toFun
+  id := @NormedAddGroupHom.id
+  comp := @NormedAddGroupHom.comp
+
+#align SemiNormedGroup.bundled_hom SemiNormedGroup.bundledHom
+
+deriving instance LargeCategory for SemiNormedGroup
+
+--Porting note: deriving fails for ConcreteCategory, adding instance manually.
+--deriving instance LargeCategory, ConcreteCategory for SemiRingCat
+instance : ConcreteCategory SemiNormedGroup := by
+  dsimp [SemiNormedGroup]
+  infer_instance
+
+--Porting Note: hasCoeToSort doesn't exist anymore
+--instance : CoeSort SemiNormedGroup (Type u) :=
+--  Bundled.hasCoeToSort
+
+instance : CoeSort SemiNormedGroup (Type _) where
+  coe X := X.α
+
+/-- Construct a bundled `SemiNormedGroup` from the underlying type and typeclass. -/
+def of (M : Type u) [SeminormedAddCommGroup M] : SemiNormedGroup :=
+  Bundled.of M
+#align SemiNormedGroup.of SemiNormedGroup.of
+
+instance (M : SemiNormedGroup) : SeminormedAddCommGroup M :=
+  M.str
+
+/-- Porting Note: Added -- needed to make coe_id work-/
+instance {X Y : SemiNormedGroup} : CoeFun (X ⟶ Y) fun _ => X → Y where
+  coe (f : X ⟶ Y) := NormedAddGroupHom.toFun f
+
+@[simp]
+theorem coe_of (V : Type u) [SeminormedAddCommGroup V] : (SemiNormedGroup.of V : Type u) = V :=
+  rfl
+#align SemiNormedGroup.coe_of SemiNormedGroup.coe_of
+
+-- Porting note : marked with high priority to short circuit simplifier's path
+@[simp (high)]
+theorem coe_id (V : SemiNormedGroup) : (𝟙 V : V → V) = id :=
+  rfl
+#align SemiNormedGroup.coe_id SemiNormedGroup.coe_id
+
+-- Porting note : marked with high priority to short circuit simplifier's path
+@[simp (high)]
+theorem coe_comp {M N K : SemiNormedGroup} (f : M ⟶ N) (g : N ⟶ K) : (f ≫ g : M → K) = g ∘ f :=
+  rfl
+#align SemiNormedGroup.coe_comp SemiNormedGroup.coe_comp
+
+instance : Inhabited SemiNormedGroup :=
+  ⟨of PUnit⟩
+
+instance ofUnique (V : Type u) [SeminormedAddCommGroup V] [i : Unique V] :
+    Unique (SemiNormedGroup.of V) :=
+  i
+#align SemiNormedGroup.of_unique SemiNormedGroup.ofUnique
+
+/--porting note: originally empty, which didn't work. Notation for composition changed?-/
+instance : Limits.HasZeroMorphisms.{u, u + 1} SemiNormedGroup where
+  Zero _ _ := NormedAddGroupHom.zero
+  comp_zero _ _ := rfl
+  zero_comp _ _ _ f:= NormedAddGroupHom.comp_zero f
+
+
+@[simp]
+theorem zero_apply {V W : SemiNormedGroup} (x : V) : (0 : V ⟶ W) x = 0 :=
+  rfl
+#align SemiNormedGroup.zero_apply SemiNormedGroup.zero_apply
+
+--/--porting note :Added, Needed to make isZero_of_subsingleton work -/
+-- instance {V W : SemiNormedGroup}: ZeroHomClass (V ⟶ W) V W where
+--   coe := NormedAddGroupHom.toFun
+--   coe_injective' := NormedAddGroupHom.coe_injective
+--   map_zero f := by simp only [NormedAddGroupHom.toFun_eq_coe, map_zero]
+
+/--porting note: Didn't work, manual rewrite-/
```

Comment:
Sorry, could you give more context in your porting notes? What didn't work?

Reply:
Some resolutions and simps didn't go through, so I made the existing proof more explicit.

Reply:
Changed the note to reflect this.